### PR TITLE
feat(send): support BIP21 URI with BOLT 11 invoice QRCode

### PR DESCRIPTION
**Pull request by krsjoseph**
#### Link this PR to an issue
Fixes #ISSUE-NUMBER

#### Type of change (Remove other not matching type)

- New feature (non-breaking change which adds functionality)

#### Describe the changes you have made in this PR -

support BIP21 URI with BOLT 11 invoice QRCode

#### How Has This Been Tested?

Not been fully as I've been having problems scanning QR codes on my laptop.

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

#### Checklist:

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by krsjoseph** [comment]
@bumi here's the PR from our paring session

File: src/app/screens/Send.tsx [hunk 1]
```diff
@@ -63,7 +75,7 @@ function Send() {
             qrbox={200}
             qrCodeSuccessCallback={(decodedText) => {
               if (invoice !== decodedText) {
-                setInvoice(decodedText);
+                checkInvoice(decodedText);
```
Comment (kiwiidb) [inline src/app/screens/Send.tsx:78]: I would name this function `checkBip21AndSetInvoice`, so it's more clear what this actually does.

**Review by kiwiidb — CHANGES_REQUESTED** [review]

File: src/app/screens/Send.tsx (hunk 1 above)
Comment (bumi) [reply src/app/screens/Send.tsx:78]: I would even do something like: 

`setInvoice(extractInvoiceFrom(decodedText))`

this way the `extractInvoiceFrom` does not need to know anything about context where it is used and just returns the invoice string if it found one. 
Potentially this makes it easier to extract and reuse somewhere else and also easier to test.

**Review by bumi — APPROVED** [review]
thanks for the PR! 
I have suggested a small refactoring, but up to you.

File: src/app/screens/Send.tsx (hunk 1 above)
Comment (krsjoseph) [reply src/app/screens/Send.tsx:78]: Thanks @bumi! I'll make these changes

Comment (bumi) [reply src/app/screens/Send.tsx:78]: (not sure if it's a good idea, was just a thought. :D )
thnks

**Review by escapedcat — APPROVED** [review]

**Pull request merged**